core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -160,26 +160,6 @@ public IndexRequest() {
     }
 
     /**
-     * Copy constructor that creates a new index request that is a copy of the one provided as an argument.
-     * The new request will inherit though headers and context from the original request that caused it.
-     */
-    public IndexRequest(IndexRequest indexRequest) {
```

Comment:
I don't think that is an argument for keeping this though - maybe an argument for reindex working another way.